client/src/components/TaskList/TaskList.js

```diff
@@ -4,26 +4,116 @@ import Paper from "@material-ui/core/Paper";
+
+    React.useEffect(() => {
+        timerIDStates.current.forEach((_, index) => {
+            if (contents[index].running && timerIDStates.current[index] === 0) {
+                timerIDStates.current[index] = setInterval(() => {
+                    setContents((prevContents) => {
+                        const newContents = ObjArrayCopy(prevContents);
+
+                        let newProgress = newContents[index].pomodoro_progress + 1 / newContents[index].pomodoro_duration * 100.0;
+                        if (newProgress > 100)
+                            newProgress = 100;
+                        else if (newProgress < 0)
+                            newProgress = 0;
+
+                        newContents[index].pomodoro_progress = newProgress;
+                        return newContents;
+                    });
+                }, 1000);
+            }
+        })
+    }, [contents, timerIDStates])
+
+    //Handler for drag end
+    const dragEndHandler = React.useCallback((result) => {
+        //Set dragging which disables animation on circular progress and fade in/out of play/pause
+        setDragging(true);
+
+        //If result is invalid, return
+        if (!result.destination || result.destination.index === result.source.index || !result) {
+            setDragging(false);
+            return;
+        }
+
+        //Swap contents
+        setContents((prevContents) => {
+            const newContents = ObjArrayCopy(prevContents);
+
+            const removedItem = { ...prevContents[result.source.index] };
+            newContents.splice(result.source.index, 1);
+            newContents.splice(result.destination.index, 0, removedItem);
+
+            //Map or foreach does not work to set timerIDStates.current. Explicit set required
+            timerIDStates.current.forEach((item, index) => {
+                clearInterval(item);
+                timerIDStates.current[index] = 0;
+            });
+
+            return newContents;
+        });
+
+        //Delay setting dragging to false to allow drop animation to complete
+        setTimeout(() => {
+            setDragging(false);
+        }, 100);
+    }, [timerIDStates]);
+
     return (
-        <>
+        <DragDropContext onDragEnd={dragEndHandler}>
             <Paper className={"main-paper"} classes={{ root: "main-paper-root" }}>
-                <Grid container>
-                    <Grid item>
-                        <Typography variant="h6">TaskList</Typography>
-                    </Grid>
-                    <Task content="Content"/>
-                    <Task content="Content"/>
-                    <Task content="Content"/>
-                    <Task content="Content"/>
-                </Grid>
-            <div className={"add-button"}>
-                <Button variant="contained" disableElevation={false}>Add</Button>
-            </div>
+                <Droppable droppableId="task-list">
+                    {(provided) => {
```

Comment:
curious, where does this come from?

Reply:
Droppable is part of the drag and drop library 
https://github.com/atlassian/react-beautiful-dnd

Is that what you're asking

Reply:
I was actually asking about the provided, but yea saw it in the documentation, thanks!